corehq/apps/hqmedia/views.py

```diff
@@ -652,7 +652,12 @@ def get(self, request, *args, **kwargs):
         data, content_type = self.multimedia.get_display_file()
         if self.thumb:
             data = CommCareImage.get_thumbnail_data(data, self.thumb)
-        return HttpResponse(data, content_type=content_type)
+        extension = ''
+        if self.multimedia.aux_media:
+            extension = self.multimedia.aux_media[-1]['uploaded_filename'].split(".")[-1]
```

Comment:
> know of any better way to do this?

No, but I'd add a method to `CommCareMultimedia` rather than digging into it's implementation details here. Maybe something like `get_file_extension()`, and make it always return a string (with a leading `.` if there is an extension, otherwise empty string) so you can change line 655 to be `extension = self.multimedia.get_file_extension()` and avoid this ugliness here. If it returns a leading `.` in the string you'll also want to change line 659.

Reply:
+1

Reply:
good point, updated